contracts/paratii/TcrPlaceholder.sol

```diff
@@ -0,0 +1,209 @@
+    uint unstakedDeposit;   // Number of tokens in the listing not locked in a challenge
+    uint challengeID;       // ID of canonical challenge
+  }
+
+  // Maps of videoId Hashes to listing data.
+  mapping(bytes32 => Listing) public listings;
+
+  ParatiiToken public token;
+  uint public minDeposit;
+  uint public applyStageLen;
+  Registry public registry;
+
+  // ---------------------------------------------------------------------------
+  // CONSTRUCTOR
+  // ------------------
+
+  function TcrPlaceholder(
+    Registry _registry,
+    address _tokenAddr,
+    uint _minDeposit,
+    uint _applyStageLen
+  ) {
+    owner = msg.sender;
+    registry = _registry;
+    token = ParatiiToken(_tokenAddr);
+    minDeposit = _minDeposit;
+    applyStageLen = _applyStageLen; // 100 blocks.
+  }
+
+  // ---------------------------------------------------------------------------
+  // Video creator functions
+  //------------------------
+
+  /**
+   * allow video creator to apply for whitelisting for a video.
+   * @param  _videoId  the videoId
+   * @param  _amount   the amount to Stake
+   */
+  function apply(string _videoId, uint _amount) external {
+    require(!isWhitelisted(_videoId));
+    require(!appWasMade(_videoId));
+    require(_amount >= minDeposit);
+
+    // set owner
+    Listing storage listing = listings[keccak256(_videoId)];
+    listing.owner = msg.sender;
+
+    // transfer tokens from user to TcrPlaceholder contract
+    require(token.transferFrom(listing.owner, this, _amount));
+
+    // sets apply stage and end time.
+    listing.applicationExpiry = block.number + applyStageLen;
+    listing.unstakedDeposit = _amount;
+
+    _Application(_videoId, _amount);
+  }
+
+  /**
+   * deposit an amount of tokens. (unstaked)
+   * @param  amount the amount to deposit
+   * @return {EVENT}      returns event log of deposit event.
+   */
+  function deposit(uint amount) external {
+
+  }
+
+  /**
+   * take out an amount of tokens from the unstaked tokens.
+   * @param  amount the amount to take out
+   */
+  function withdraw(uint amount) external {
+
+  }
+
+  /**
+   * allow video creator to remove a video from the listing and return tokens.
+   * @param  _videoId videoId to remove
+   */
+  function exit(string _videoId) external {
+    Listing storage listing = listings[keccak256(_videoId)];
+
+    require(msg.sender == listing.owner);
+    require(isWhitelisted(_videoId));
+
+    // Cannot exit during ongoing challenge
+    /* require(!challenges[listing.challengeID].isInitialized() ||
+            challenges[listing.challengeID].isResolved()); */
+
+    // Remove domain & return tokens
+    resetListing(_videoId);
+  }
+
+  /**
+   * remove a video , Only the TCR owner can do this.
+   * @param  _videoId id of the video to remove.
+   */
+  function removeListing(string _videoId) public onlyOwner {
+    Listing storage listing = listings[keccak256(_videoId)];
+    require(isWhitelisted(_videoId));
```

Comment:
I think the owner should be able to remove any application, whether it is already whitelisted or not